Subcategories / parent categories not loading

Spinning forever.
Latest alpha

Error has a super long log:

```
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment: Error occurred while loading queried subcategories
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.length()' on a null object reference
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at java.io.StringReader.<init>(StringReader.java:50)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.text.HtmlToSpannedConverter.convert(Html.java:746)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.text.Html.fromHtml(Html.java:242)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.text.Html.fromHtml(Html.java:196)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at fr.free.nrw.commons.utils.StringUtils.getParsedStringFromHtml(StringUtils.java:9)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at fr.free.nrw.commons.category.CategoryImageUtils.getCreator(CategoryImageUtils.java:111)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at fr.free.nrw.commons.category.CategoryImageUtils.getMediaFromPage(CategoryImageUtils.java:68)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at fr.free.nrw.commons.category.CategoryImageUtils.getSubCategoryList(CategoryImageUtils.java:49)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at fr.free.nrw.commons.mwapi.ApacheHttpClientMediaWikiApi.getSubCategoryList(ApacheHttpClientMediaWikiApi.java:645)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at fr.free.nrw.commons.category.SubCategoryListFragment.lambda$initSubCategoryList$1$SubCategoryListFragment(SubCategoryListFragment.java:100)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at fr.free.nrw.commons.category.SubCategoryListFragment$$Lambda$1.call(Unknown Source:2)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.internal.operators.observable.ObservableFromCallable.subscribeActual(ObservableFromCallable.java:42)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.Observable.subscribe(Observable.java:12030)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at com.tspoon.traceur.ObservableOnAssemblyCallable.subscribeActual(ObservableOnAssemblyCallable.java:42)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.Observable.subscribe(Observable.java:12030)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:579)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at java.util.concurrent.FutureTask.run(FutureTask.java:266)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at java.lang.Thread.run(Thread.java:764)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment: Caused by: com.tspoon.traceur.TraceurException: Debug Exception generated at call site
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at dalvik.system.VMStack.getThreadStackTrace(Native Method)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at java.lang.Thread.getStackTrace(Thread.java:1538)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.Observable.fromCallable(Observable.java:1721)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at fr.free.nrw.commons.category.SubCategoryListFragment.initSubCategoryList(SubCategoryListFragment.java:100)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at fr.free.nrw.commons.category.SubCategoryListFragment.onCreateView(SubCategoryListFragment.java:75)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.app.Fragment.performCreateView(Fragment.java:2346)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1428)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1759)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1827)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.app.BackStackRecord.executeOps(BackStackRecord.java:797)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.app.FragmentManagerImpl.executeOps(FragmentManager.java:2596)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.app.FragmentManagerImpl.executeOpsTogether(FragmentManager.java:2383)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManager.java:2338)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.app.FragmentManagerImpl.execSingleAction(FragmentManager.java:2215)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.app.BackStackRecord.commitNowAllowingStateLoss(BackStackRecord.java:649)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.app.FragmentPagerAdapter.finishUpdate(FragmentPagerAdapter.java:145)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.view.ViewPager.populate(ViewPager.java:1238)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.view.ViewPager.populate(ViewPager.java:1086)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.view.ViewPager.onMeasure(ViewPager.java:1616)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.View.measure(View.java:23169)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.widget.RelativeLayout.measureChildHorizontal(RelativeLayout.java:715)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.widget.RelativeLayout.onMeasure(RelativeLayout.java:461)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.View.measure(View.java:23169)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.widget.DrawerLayout.onMeasure(DrawerLayout.java:1059)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.View.measure(View.java:23169)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6749)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v7.widget.ContentFrameLayout.onMeasure(ContentFrameLayout.java:141)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.View.measure(View.java:23169)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6749)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1535)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.widget.LinearLayout.measureVertical(LinearLayout.java:825)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.widget.LinearLayout.onMeasure(LinearLayout.java:704)
2018-12-17 16:07:35.622 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.View.measure(View.java:23169)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6749)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.View.measure(View.java:23169)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6749)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1535)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.widget.LinearLayout.measureVertical(LinearLayout.java:825)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.widget.LinearLayout.onMeasure(LinearLayout.java:704)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.View.measure(View.java:23169)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6749)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at com.android.internal.policy.DecorView.onMeasure(DecorView.java:716)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.View.measure(View.java:23169)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:2718)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:1572)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1855)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1460)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7183)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.Choreographer$CallbackRecord.run(Choreographer.java:949)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.Choreographer.doCallbacks(Choreographer.java:761)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.Choreographer.doFrame(Choreographer.java:696)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:935)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.os.Handler.handleCallback(Handler.java:873)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.os.Handler.dispatchMessage(Handler.java:99)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.os.Looper.loop(Looper.java:193)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.app.ActivityThread.main(ActivityThread.java:6669)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at java.lang.reflect.Method.invoke(Native Method)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment: Caused by: com.tspoon.traceur.TraceurException: Debug Exception generated at call site
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at dalvik.system.VMStack.getThreadStackTrace(Native Method)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at java.lang.Thread.getStackTrace(Thread.java:1538)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.Observable.subscribeOn(Observable.java:12107)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at fr.free.nrw.commons.category.SubCategoryListFragment.initSubCategoryList(SubCategoryListFragment.java:101)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment: 	... 58 more
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment: Caused by: com.tspoon.traceur.TraceurException: Debug Exception generated at call site
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at dalvik.system.VMStack.getThreadStackTrace(Native Method)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at java.lang.Thread.getStackTrace(Thread.java:1538)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.Observable.observeOn(Observable.java:9756)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.Observable.observeOn(Observable.java:9689)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at fr.free.nrw.commons.category.SubCategoryListFragment.initSubCategoryList(SubCategoryListFragment.java:102)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment: 	... 58 more
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment: Caused by: com.tspoon.traceur.TraceurException: Debug Exception generated at call site
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at dalvik.system.VMStack.getThreadStackTrace(Native Method)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at java.lang.Thread.getStackTrace(Thread.java:1538)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.Observable.timeout0(Observable.java:13568)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.Observable.timeout(Observable.java:13398)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at fr.free.nrw.commons.category.SubCategoryListFragment.initSubCategoryList(SubCategoryListFragment.java:103)
2018-12-17 16:07:35.623 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment: 	... 58 more
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment: Error occurred while loading queried parentcategories
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.length()' on a null object reference
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at java.io.StringReader.<init>(StringReader.java:50)
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.text.HtmlToSpannedConverter.convert(Html.java:746)
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.text.Html.fromHtml(Html.java:242)
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.text.Html.fromHtml(Html.java:196)
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at fr.free.nrw.commons.utils.StringUtils.getParsedStringFromHtml(StringUtils.java:9)
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at fr.free.nrw.commons.category.CategoryImageUtils.getCreator(CategoryImageUtils.java:111)
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at fr.free.nrw.commons.category.CategoryImageUtils.getMediaFromPage(CategoryImageUtils.java:68)
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at fr.free.nrw.commons.category.CategoryImageUtils.getSubCategoryList(CategoryImageUtils.java:49)
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at fr.free.nrw.commons.mwapi.ApacheHttpClientMediaWikiApi.getParentCategoryList(ApacheHttpClientMediaWikiApi.java:685)
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at fr.free.nrw.commons.category.SubCategoryListFragment.lambda$initSubCategoryList$2$SubCategoryListFragment(SubCategoryListFragment.java:106)
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at fr.free.nrw.commons.category.SubCategoryListFragment$$Lambda$4.call(Unknown Source:2)
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.internal.operators.observable.ObservableFromCallable.subscribeActual(ObservableFromCallable.java:42)
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.Observable.subscribe(Observable.java:12030)
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at com.tspoon.traceur.ObservableOnAssemblyCallable.subscribeActual(ObservableOnAssemblyCallable.java:42)
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.Observable.subscribe(Observable.java:12030)
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:579)
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at java.util.concurrent.FutureTask.run(FutureTask.java:266)
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at java.lang.Thread.run(Thread.java:764)
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment: Caused by: com.tspoon.traceur.TraceurException: Debug Exception generated at call site
2018-12-17 16:07:35.769 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at dalvik.system.VMStack.getThreadStackTrace(Native Method)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at java.lang.Thread.getStackTrace(Thread.java:1538)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.Observable.fromCallable(Observable.java:1721)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at fr.free.nrw.commons.category.SubCategoryListFragment.initSubCategoryList(SubCategoryListFragment.java:106)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at fr.free.nrw.commons.category.SubCategoryListFragment.onCreateView(SubCategoryListFragment.java:75)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.app.Fragment.performCreateView(Fragment.java:2346)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1428)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1759)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1827)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.app.BackStackRecord.executeOps(BackStackRecord.java:797)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.app.FragmentManagerImpl.executeOps(FragmentManager.java:2596)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.app.FragmentManagerImpl.executeOpsTogether(FragmentManager.java:2383)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManager.java:2338)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.app.FragmentManagerImpl.execSingleAction(FragmentManager.java:2215)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.app.BackStackRecord.commitNowAllowingStateLoss(BackStackRecord.java:649)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.app.FragmentPagerAdapter.finishUpdate(FragmentPagerAdapter.java:145)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.view.ViewPager.populate(ViewPager.java:1238)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.view.ViewPager.populate(ViewPager.java:1086)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.view.ViewPager.onMeasure(ViewPager.java:1616)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.View.measure(View.java:23169)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.widget.RelativeLayout.measureChildHorizontal(RelativeLayout.java:715)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.widget.RelativeLayout.onMeasure(RelativeLayout.java:461)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.View.measure(View.java:23169)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v4.widget.DrawerLayout.onMeasure(DrawerLayout.java:1059)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.View.measure(View.java:23169)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6749)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.support.v7.widget.ContentFrameLayout.onMeasure(ContentFrameLayout.java:141)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.View.measure(View.java:23169)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6749)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1535)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.widget.LinearLayout.measureVertical(LinearLayout.java:825)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.widget.LinearLayout.onMeasure(LinearLayout.java:704)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.View.measure(View.java:23169)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6749)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.View.measure(View.java:23169)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6749)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1535)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.widget.LinearLayout.measureVertical(LinearLayout.java:825)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.widget.LinearLayout.onMeasure(LinearLayout.java:704)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.View.measure(View.java:23169)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6749)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at com.android.internal.policy.DecorView.onMeasure(DecorView.java:716)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.View.measure(View.java:23169)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:2718)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:1572)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1855)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1460)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7183)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.Choreographer$CallbackRecord.run(Choreographer.java:949)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.Choreographer.doCallbacks(Choreographer.java:761)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.Choreographer.doFrame(Choreographer.java:696)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:935)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.os.Handler.handleCallback(Handler.java:873)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.os.Handler.dispatchMessage(Handler.java:99)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.os.Looper.loop(Looper.java:193)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at android.app.ActivityThread.main(ActivityThread.java:6669)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at java.lang.reflect.Method.invoke(Native Method)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment: Caused by: com.tspoon.traceur.TraceurException: Debug Exception generated at call site
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at dalvik.system.VMStack.getThreadStackTrace(Native Method)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at java.lang.Thread.getStackTrace(Thread.java:1538)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.Observable.subscribeOn(Observable.java:12107)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at fr.free.nrw.commons.category.SubCategoryListFragment.initSubCategoryList(SubCategoryListFragment.java:107)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment: 	... 58 more
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment: Caused by: com.tspoon.traceur.TraceurException: Debug Exception generated at call site
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at dalvik.system.VMStack.getThreadStackTrace(Native Method)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at java.lang.Thread.getStackTrace(Thread.java:1538)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.Observable.observeOn(Observable.java:9756)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.Observable.observeOn(Observable.java:9689)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at fr.free.nrw.commons.category.SubCategoryListFragment.initSubCategoryList(SubCategoryListFragment.java:108)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment: 	... 58 more
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment: Caused by: com.tspoon.traceur.TraceurException: Debug Exception generated at call site
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at dalvik.system.VMStack.getThreadStackTrace(Native Method)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at java.lang.Thread.getStackTrace(Thread.java:1538)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.Observable.timeout0(Observable.java:13568)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at io.reactivex.Observable.timeout(Observable.java:13398)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment:     at fr.free.nrw.commons.category.SubCategoryListFragment.initSubCategoryList(SubCategoryListFragment.java:109)
2018-12-17 16:07:35.770 5255-5255/fr.free.nrw.commons E/SubCategoryListFragment: 	... 58 more
```

--- a/app/src/main/java/fr/free/nrw/commons/category/CategoryImageUtils.java
+++ b/app/src/main/java/fr/free/nrw/commons/category/CategoryImageUtils.java
@@ -26,9 +26,11 @@ public class CategoryImageUtils {
      */
     public static List<Media> getMediaList(NodeList childNodes) {
         List<Media> categoryImages = new ArrayList<>();
+
         for (int i = 0; i < childNodes.getLength(); i++) {
             Node node = childNodes.item(i);
-            if (getMediaFromPage(node).getFilename().substring(0,5).equals("File:")){
+
+            if (getFileName(node).substring(0, 5).equals("File:")) {
                 categoryImages.add(getMediaFromPage(node));
             }
         }
@@ -46,7 +48,7 @@ public class CategoryImageUtils {
         List<String> subCategories = new ArrayList<>();
         for (int i = 0; i < childNodes.getLength(); i++) {
             Node node = childNodes.item(i);
-            subCategories.add(getMediaFromPage(node).getFilename());
+            subCategories.add(getFileName(node));
         }
         Collections.sort(subCategories);
         return subCategories;

